File: src/javascript/image2image/stable_diffusion.js
import fs from "node:fs";
import axios from "axios";
import FormData from "form-data";

const formData = {
  prompt:
    "Astronaut in a jungle, cold color palette, muted colors, detailed, 8k",
  output_format: "jpeg",
  model: "sd3", // sd3 or sd3-turbo turboは高速で低価格
  mode: "image-to-image",
  strength: 0.6, // 0だと入力画像そのまま、1だと完全に生成画像
  image: fs.createReadStream("./assets/sample.png"),
};

const response = await axios.postForm(
  `https://api.stability.ai/v2beta/stable-image/generate/sd3`,
  axios.toFormData(formData, new FormData()),
  {
    validateStatus: undefined,
    responseType: "arraybuffer",
    headers: {
      Authorization: `Bearer ${process.env.STABILITY_API_KEY}`,
      Accept: "image/*",
    },
  }
);

if (response.status === 200) {
  fs.writeFileSync(
    "./src/javascript/image2image/result.png",
    Buffer.from(response.data)
  );
} else {
  throw new Error(`${response.status}: ${response.data.toString()}`);
}

File: sample/python/image2image/stable_diffusion.py
import requests
import os
from PIL import Image


def get_image_bytes(image_path: str) -> bytes:
    with open(image_path, "rb") as image_file:
        return image_file.read()


API_KEY = os.getenv("STABILITY_API_KEY")

response = requests.post(
    f"https://api.stability.ai/v2beta/stable-image/generate/sd3",
    headers={
        "authorization": f"Bearer {API_KEY}",
        "accept": "image/*",
    },
    files={"image": get_image_bytes("./assets/sample.png")},
    data={
        "prompt": "Astronaut in a jungle, cold color palette, muted colors, detailed, 8k",  # SDでは英語である必要がある
        "output_format": "png",
        "model": "sd3",  # sd3 or sd3-turbo turboは高速で低価格
        "mode": "image-to-image",
        "strength": 0.6,  # 0だと入力画像そのまま、1だと完全に生成画像
    },
)

if response.status_code == 200:
    with open("./src/python/image2image/sd3_result.jpeg", "wb") as file:
        file.write(response.content)
else:
    raise Exception(str(response.json()))
